src/main/java/com/github/fge/jsonpatch/CopyOperation.java

```diff
@@ -40,24 +40,19 @@
  *
  * <p>It is an error if {@code from} fails to resolve to a JSON value.</p>
  */
-public final class CopyOperation
-    extends DualPathOperation
-{
+public final class CopyOperation extends DualPathOperation {
+
     @JsonCreator
-    public CopyOperation(@JsonProperty("from") final JsonPointer from,
-        @JsonProperty("path") final JsonPointer path)
-    {
+    public CopyOperation(@JsonProperty("from") final String from, @JsonProperty("path") final String path) {
         super("copy", from, path);
     }
 
     @Override
-    public JsonNode apply(final JsonNode node)
-        throws JsonPatchException
-    {
-        final JsonNode dupData = from.path(node).deepCopy();
-        if (dupData.isMissingNode())
-            throw new JsonPatchException(BUNDLE.getMessage(
-                "jsonPatch.noSuchPath"));
+    public JsonNode apply(final JsonNode node) throws JsonPatchException {
+        String jsonPath = JsonPathParser.tmfStringToJsonPath(from);
```

Comment:
Here is no `final` keyword, line below (and other places) it is. Is there a specific reason for that? I can live with no `final` in front of local vars.

Reply:
No, it was just me being a bit frivolous with `final`, fixed